src/Compilers/Server/VBCSCompilerTests/CompilerServerTests.cs

```diff
@@ -307,7 +309,7 @@ private void VerifyResultAndOutput((int ExitCode, string Output) result, TempDir
                     shouldRunOnServer: false);
 
                 Assert.Equal(result.ExitCode, 1);
-                Assert.Equal(@"test.vb(1) : error BC30037: Character is not valid.
+                Assert.Equal($@"test.vb(1) : error BC30037: {ReplaceNonASCIICharacters(VBResources.ERR_IllegalChar)}
```

Comment:
❔ Why is the call to `ReplaceNonASCIICharacters` necessary?

➡️ The use of `ERR_IllegalChar` is intended to avoid the need to write non-ASCII characters to the output. Ironically, this does not work as expected due to localization.

Reply:
The problem is that the error message contains non ascii characters ("Ungültiges Zeichen."). So because the pipe is set to ASCII the "ü" is replaced by "?".

The desireable output for german would probably be "Ungueltiges Zeichen." (replacing ü by ue) or maybe use a different word without them, maybe "Unerlaubtes Zeichen", but I dont think you can do this for every language. For languages with a completely different character set (e.g. chinese) this test becomes completely useless.